java/vector/src/main/java/org/apache/arrow/vector/BaseFixedWidthVector.java

```diff
@@ -393,13 +393,9 @@ public Field getField() {
   public ArrowBuf[] getBuffers(boolean clear) {
     final ArrowBuf[] buffers;
     setReaderAndWriterIndex();
-    if (getBufferSize() == 0) {
```

Comment:
I believe this if check came from an earlier refactor that happened in Arrow and several tests had failed in Dremio. I think we should keep it.

Reply:
In the old implementation, VectorUnloader.java checks the typeBufferCount with getFieldBuffers, it's ok since getFieldBuffers returns all buffers without check buffer size. If we keep this if, then the expected buffer count check in VectorUnloader is invalid, should we remove that check directly?

Reply:
> In the old implementation, VectorUnloader.java checks the typeBufferCount with getFieldBuffers, it's ok since getFieldBuffers returns all buffers without check buffer size. If we keep this if, then the expected buffer count check in VectorUnloader is invalid, should we remove that check directly?

I am okay with this change but it would be great if someone from Dremio can bless this (based on my previous comment)

Reply:
If we keep this 'if' here, then I guess we should make some change in VectorUnloader.
If vector.getBufferSize ==0, then no buffers would be sent via IPC, however, the VectorLoader depends on TypeBufferCount to decide how many buffers to load into a vector, and in this case, something is wrong.
To solve this, we may add a check in VectorUnloader, if vector bufferSize==0, we should append it's field buffers(by call getFieldBuffers) also, even their writerIndex/readerIndex=0. In this way, we could keep the 'if (getBufferSize() == 0)' in vectors and the IPC also works well.

Do you think we should keep this PR or update it as I suggested above?